packages/terra-application/src/terra-dev-site/doc/application/ApplicationLoadingOverlay.md

```diff
@@ -0,0 +1,39 @@
+# Terra Application
+
+## Application Loading Overlay
+
+The ApplicationLoadingOverlay allows for the presentation of loading overlays within the Terra Application framework.
+
+Mounting an open ApplicationLoadingOverlay will result in the framework rendering a loading overlay. Unmounting the ApplicationLoadingOverlay (or removing the `isOpen` prop) will subsequently remove the loading overlay.
+
+### ApplicationLoadingOverlay Props
```

Comment:
could we render these props in the template so we don't have to worry about these get missed/out-of-date when code changes occur?

Reply:
We could if react-docgen was smarter. Because its a functional component that returns null, react-docgen doesn't recognize the component and cannot generate a props-table for it. I didn't feel like rewriting the component to be a class to work around this, but maybe this will work better with a future version of react-docgen?

Reply:
Could you log your props table enhancement requests to the terra-dev-site repo?

Reply:
@tbiethman I got around this by moving my propTypes to it's own file and then importing it as a codeblock with the new codeblock loader: 
```
import PropsTable from '.../../applicationLoadingOverlayPropTypes?dev-site-codeblock';
```
Similar to what we do in terra-dev-site for it's config: https://engineering.cerner.com/terra-dev-site/getting-started/terra-dev-site/site-config/site-config

Reply:
I'm going to be circling back around on the docs here in a future PR; I'll cover this scenario with the holistic update to mdx.